### ISSUE TITLE
Pip 18.1 traces in the 3.6.10-alpine3.11 image

### ISSUE THREAD

**Issue by ghevge** [docker-library/python#494]
Hi,
The security code scanners we are using are reporting  the existence of Pip 18.1 in the base image.
Will you be able to clean it up as there seem to be problems with this particular pip version ?

Thanks
G
![1](https://user-images.githubusercontent.com/42102616/83579232-b39cb580-a506-11ea-9449-2d0d2cde1265.jpg)
![2](https://user-images.githubusercontent.com/42102616/83579235-b4cde280-a506-11ea-9ada-83a98a2fbceb.jpg)
![3](https://user-images.githubusercontent.com/42102616/83579237-b4cde280-a506-11ea-9c1f-0ef165f21f05.jpg)

**Comment by wglambert** [comment]
https://github.com/docker-library/python/blob/208a857d963b7310cde11843f0ed8c2c9712dc09/3.6/alpine3.11/Dockerfile#L147

```console
$ docker run -it --rm python:3.6.10-alpine3.11
Unable to find image 'python:3.6.10-alpine3.11' locally
3.6.10-alpine3.11: Pulling from library/python
cbdbe7a5bc2a: Pull complete 
26ebcd19a4e3: Pull complete 
170f815b6f8f: Pull complete 
8577981c6bd3: Pull complete 
05fa7f8946aa: Pull complete 
Digest: sha256:cfde075076d89a4b3f79933c9cc281900dcd0e6a3deb01ffd3a4ac371fab60c8
Status: Downloaded newer image for python:3.6.10-alpine3.11

/ # pip --version
pip 20.1.1 from /usr/local/lib/python3.6/site-packages/pip (python 3.6)
```

**Comment by ghevge** [comment]
I agree with you. I've tried the same thing locally and i get the same result. Still xray scanner find those pip 18.1 traces in the image. The sha highlighted in the attached image points to this line:


RUN set -ex \
	&& apk add --no-cache --virtual .fetch-deps \
		gnupg \
		tar \
		xz \
	\

**Comment by tianon** [comment]
It might be detecting the version of `pip` that comes bundled with our Python release, but AFAIK, that doesn't end up in the layer anywhere since we do `--without-ensurepip`:

https://github.com/docker-library/python/blob/208a857d963b7310cde11843f0ed8c2c9712dc09/3.6/alpine3.11/Dockerfile#L78

**Comment by ghevge** [comment]
Isn't it possible to get it with one of those dependencies ?

**Comment by yosifkit** [comment]
> The "--without-ensurepip" flag is there to disable running it at install time. Fully removing the ensurepip module is not a supported configuration - if a distro wants to do that, they're going to have to carry a patch.
>
> https://bugs.python.org/issue20417

> Even if pip is made available globally by other means, do not remove the ensurepip module in Python 3.4 or later.
>
> https://www.python.org/dev/peps/pep-0453/#recommendations-for-downstream-distributors

The pip bundled in the ensurepip module is unsued. So, working as designed.

**Comment by ghevge** [comment]
pip 18.1 is probably coming from the alpine 3.11 image: http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz.

Any plan to release python3.6 alpine 3.12 any time soon ? From what i can see pip 18.1 was removed from 3.12

**Comment by tianon** [comment]
Pip 18.1 is not coming from Alpine -- it's coming from the version of Pip that's embedded in Python itself as part of the `ensurepip` module that @yosifkit mentioned above.  For us to update it will require a newer release of Python.

Alpine 3.12 variants were added in https://github.com/docker-library/python/pull/493.

**Comment by ghevge** [comment]
For the record, the way i fixed my problem was by removing the pip-18.1-py2.py3-none-any.whl file from the python folder

rm /usr/local/lib/python3.6/ensurepip/_bundled/pip-18.1-py2.py3-none-any.whl
